### ISSUE TITLE
version 1.0 ?

### ISSUE THREAD

**Issue by stepan-tsirkin** [wannier-berri/wannier-berri#61]
Hi guys, 
speaking about versions, so far I was quite chaotic in incrementing the sub- or susub-version digit, but it is coming to the point to put `1.0.0`

What do you think, what are the milestones that need to be passed to do this. I think, merging  Ray #52 is one of them, because it makes WannierBerri really massively parallel, which has been the biggest drawback compared to postw90. 

Another big update coming soon is the [formula](/wannier-berri/wannier-berri/tree/formula) , which among others will introduce tetrahedron integration method, and some other features..

@jaemolihm , @Liu-Xiaoxiong  , what do you think?

**Comment by jaemolihm** [comment]
I also think it is a good idea! It's great that WannierBerri is close to v1.0.

One convention about a major version is that the APIs (for our case, the arguments of `system`, `integrate`, `tabulate`, `grid`, etc.) should be stable enough. I think they are already quite stable, so it is reasonable to go to v1.0.

**Comment by stepan-tsirkin** [comment]
> stable **enough**
Of course we will not fix them totally stable in the nearest future. 

One thing that I want to change is to create a class Parallel, that will store all info needed for parallelisation. Then it will be a single parallelisatoion-related parameter for `integrate` and `tabulate` Otherwise this number of parameters is growing fast. With the same motivation I earlier introduced the Grid class. 

Then, I hope that for some time we will only add parameters, not much removing/renaming them.

**Comment by jaemolihm** [comment]
Another thing to do is to document all quantities implemented and add tests for all of them.

Also, I wonder whether the Fermi sea and Fermi ocean are just different implementations of the same thing. If this is the case, it may be possible to deprecate (remove) one before 1.0.

**Comment by stepan-tsirkin** [comment]
> Another thing to do is to document all quantities implemented and add tests for all of them. 

Right, can we make a list of what is not documented yet? 

Regarding tests, the big part that is notcovered by tests at all is tabuation and plotting. However, I am not sure how one can write tests for the plotting (the part involving matplotlib) Or maybe, it is not so necessary. In my opinion, the code should primarily produce data. The plotting routines are a bonus, nothing more.

> Also, I wonder whether the Fermi sea and Fermi ocean are just different implementations of the same thing. If this is the case, it may be possible to deprecate (remove) one before 1.0.

Yes, they are the same, and Femi ocean performs much better for the quantities requiring summation over 3 band indices (like the Berry dipole), but yet not sure about 2-band summation (like AHC). But Fermi ocean allows an easier implementation of new quantities, and hence should be more stable, so it will be the future. I think, when we move everything from "sea" to "ocean", the "sea" should be removed.

**Comment by stepan-tsirkin** [comment]
regarding removing old things, also there is a question: https://github.com/wannier-berri/wannier-berri/pull/52#issuecomment-850830693

**Comment by jaemolihm** [comment]
> Right, can we make a list of what is not documented yet?

I made a separate issue: #62

**Comment by jaemolihm** [comment]
kBT is in eV/kB, while smearEf is in Kelvin. I think it would be better to make all temperature parameters in Kelvin.

**Comment by stepan-tsirkin** [comment]
@jaemolihm @Liu-Xiaoxiong @manxkim @ivosouza0 
I would like to make a release to pypi, after merge of #64 .  Because there is a lot new developments that the users can enjoy right now, although not all features are well documented yet.

Also, because we did not make releases for halt-year already, we goes away from the earlier spirit to release every new development. 

The only question, should I already give it a number smth like 1.0.0a0, or should it still be 0.10.0  . The API has changed, but it will change further, I think. 
In any case, I like that development of WannierBerri goes continuously, without any special milestones or stops, so I plan to resume tradition to make releases several times per month. 

Or maybe we want to make the release of version-1.0.0 an "event", with a zoom presentation and maybe a tutorial and/or a Q&A session?  We can make it as an independent online event, or make it at the Wannier school in Trieste.  What do you think?

**Comment by ivosouza0** [comment]
On Tue, 2 Nov 2021, Jae-Mo Lihm wrote:

> I do not have a strong opinion, but I like the idea of making the 
> release of version-1.0.0 an "event". For that, we may make release 
> 0.10.0 now. We can also keep updating wannierberri (to 0.11, 0.12, 
> ...) until the "event".

Yes, this sounds like a good idea.

   Ivo

**Comment by jaemolihm** [comment]
As branch `version-1.0.0` is merged to master, can we delete the following branches? `formula+berry-boltzmann`, `branch-0.9.x`, `version-1.0.0`, `Faraday`, `formula`

**Comment by stepan-tsirkin** [comment]
Yes, we can, I removed them now.

**Comment by stepan-tsirkin** [comment]
@jaemolihm , could you explain for the dummies, what should be the proper strategy with releases? i.e.  maybe in future we should first create a release on github, and then upload it to pypi? Or can this be automatically connected?
Or not, because to pypi we upload only the code, not the tests/tutorials

**Comment by stepan-tsirkin** [comment]
Also, does it make sense to keep all releases, including minor bugfixes ?

**Comment by jaemolihm** [comment]
First, I should say that I also have little experience in releases.

Frankly, I did not intend to make a GitHub release... I just wanted to add a git tag, but I mistakenly made a release. As we are using pypi, I think we do not need to use GitHub release. I deleted the one I created.

But git tags are useful, as it tells you which git commit was used in the pypi release. I added a script to automatize adding and pushing git tag in uploadpypi.sh: #87
